benchmarks/functional_autograd_benchmark/functional_autograd_benchmark.py

```diff
@@ -121,16 +195,18 @@ def noop():
     model, inp = model_getter(device)
 
     v = get_v_for(model, inp, task)
-    # Warmup
-    run_once(model, inp, task, v)
 
-    elapsed = []
-    for it in range(args.num_iters):
-        do_sync()
-        start = time.time()
-        run_once(model, inp, task, v)
-        do_sync()
-        elapsed.append(time.time() - start)
+    with torch.set_grad_enabled(run_once_fn == run_once):
```

Comment:
@zou3519 if functorch => set grad enabled false and otherwise true. I hope this should do what we want.

Reply:
I think the set_grad_enabled shouldn't be in this function (run_model). Here are two reasons:
- run_model shouldn't need to worry about what kind of function it got passed.
- In the future, I can see a case where we benchmark autograd.functional with create_graph=True v.s. functorch transforms. In that case, the current approach (this set_grad_enabled) is too big of a hammer.

Here's a proposal: we can change get_task_functorch to look like the following:
```
def get_task_functorch(task: str) -> Callable:
    @torch.no_grad()
    def vjp(model, inp, v=None, strict=None):
        ...

    @torch.no_grad()
    def jvp(model, inp, v=None, strict=None):
        ...
    
    ...
```
This way, in the future, we might come up with a "jvp_create_graph" task and we have a way to say that the functorch equivalent of the task is jvp but without the torch.no_grad.

Reply:
Sounds good. Let me update it this way